third_party/kythe/upgrade_from_kythe.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#
+# Upgrades the third_party/kythe directory to a version in a local Kythe repository.
+
+set -ux
+
+usage() {
+  echo "Usage: $(basename "$0") <kythe-repository-root>" >&2
+  exit 1
+}
+
+case $# in
+  1)
+    KYTHE_REPO="$(readlink -e "$1")" ;;
+  *)
+    usage ;;
+esac
+
+cd "$(dirname "$0")"/../..
+
+echo "Kythe repository: $KYTHE_REPO" >&2
+
+COMMIT=$(git --git-dir="$KYTHE_REPO/.git" rev-parse HEAD)
+echo "Upgrading third_party/kythe to version $COMMIT" >&2
+
+kythe_paths=(
+ "proto"
+ "go/platform/analysis"
+ "go/platform/delimited"
+ "go/platform/kindex"
+ "java/com/google/devtools/kythe/common"
+ "java/com/google/devtools/kythe/extractors/shared"
+ "java/com/google/devtools/kythe/platform/shared"
+ "java/com/google/devtools/kythe/platform/java"
+ "java/com/google/devtools/kythe/platform/java/filemanager"
+ )
+
+for path in "${kythe_paths[@]}"; do
+  rm -rf third_party/kythe/$path
```

Comment:
This will miss removing directories that are removed from $kythe_paths.

Using `rysync --delete` with either `--include-from` or `--files-from` will make the import more maintainable.